src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -362,95 +364,117 @@ MultiPlanTree(Query *queryTree)
 static void
 ErrorIfQueryNotSupported(Query *queryTree)
 {
-	char *errorDetail = NULL;
+	char *errorMessage = NULL;
 	bool hasTablesample = false;
 	bool hasUnsupportedJoin = false;
 	bool hasComplexJoinOrder = false;
 	bool hasComplexRangeTableType = false;
 	bool preconditionsSatisfied = true;
+	const char *errorDetail = NULL;
+	const char *joinDetail = "Consider joining tables on partition column and have "
+							 "equal filter on joining columns.";
+	const char *filterDetail = "Consider using an equality filter on the distributed "
+							   "table's partition column.";
 
 	if (queryTree->hasSubLinks)
 	{
 		preconditionsSatisfied = false;
-		errorDetail = "Subqueries other than in from-clause are currently unsupported";
+		errorMessage = "cannot run distributed queries with subqueries other than "
```

Comment:
_other than in_ reads a little informal. Let's do _outside the FROM clause_ (_from-clause_ is also weird).